crates/op-rpc-types/src/op/block.rs

```diff
@@ -0,0 +1,144 @@
+//! Block RPC types.
+
+#![allow(unknown_lints, non_local_definitions)]
+
+use crate::op::transaction::Transaction;
+use alloy::rpc::types::eth::{
+    BlockTransactionHashes, BlockTransactionHashesMut, Header, Rich, Withdrawal,
+};
+use alloy_primitives::{B256, U256};
+use serde::{Deserialize, Serialize};
+
+/// Block representation
+#[derive(Default, Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Block {
+    /// Header of the block.
+    #[serde(flatten)]
+    pub header: Header,
+    /// Uncles' hashes.
+    #[serde(default)]
+    pub uncles: Vec<B256>,
+    /// Block Transactions. In the case of an uncle block, this field is not included in RPC
+    /// responses, and when deserialized, it will be set to [BlockTransactions::Uncle].
+    #[serde(
+        default = "BlockTransactions::uncle",
+        skip_serializing_if = "BlockTransactions::is_uncle"
+    )]
+    pub transactions: BlockTransactions,
+    /// Integer the size of this block in bytes.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub size: Option<U256>,
+    /// Withdrawals in the block.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub withdrawals: Option<Vec<Withdrawal>>,
+}
```

Comment:
This should just be a re-export of the default json rpc block, no?

Reply:
The BlockTransactions struct has changed to include the optimism transactions, so I included the block here to use the new BlockTransactions

Reply:
we should make that struct take a generic